File: src/main/java/de/tectoast/emolga/commands/flo/EmoteStealCommand.java
package de.tectoast.emolga.commands.flo;

import de.tectoast.emolga.commands.Command;
import de.tectoast.emolga.commands.CommandCategory;
import de.tectoast.emolga.commands.GuildCommandEvent;
import net.dv8tion.jda.api.entities.TextChannel;

public class EmoteStealCommand extends Command {


    public EmoteStealCommand() {
        super("emotesteal", "Stealt Emotes... lol", CommandCategory.Flo);
        setArgumentTemplate(ArgumentManagerTemplate.noArgs());
    }

    @Override
    public void process(GuildCommandEvent e) {
        TextChannel tco = e.getChannel();
        if (emoteSteal.remove(tco.getIdLong())) {
            tco.sendMessage("Der EmoteSteal wurde deaktiviert!").queue();
        } else {
            emoteSteal.add(tco.getIdLong());
            tco.sendMessage("Der EmoteSteal wurde aktiviert!").queue();
        }
    }
}

File: src/main/kotlin/de/tectoast/emolga/commands/flo/EmoteStealCommand.kt
package de.tectoast.emolga.commands.flo

import de.tectoast.emolga.commands.Command
import de.tectoast.emolga.commands.CommandCategory
import de.tectoast.emolga.commands.GuildCommandEvent

class EmoteStealCommand : Command("emotesteal", "Stealt Emotes... lol", CommandCategory.Flo) {
    init {
        argumentTemplate = ArgumentManagerTemplate.noArgs()
    }

    override suspend fun process(e: GuildCommandEvent) {
        val tco = e.textChannel
        emoteSteal.remove(tco.idLong).also {
            if (!it) emoteSteal.add(tco.idLong)
            e.reply("Der EmoteSteal wurde ${if (it) "de" else ""}aktiviert!")
        }
    }
}